keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/binary/DivisionVertex.java

```diff
@@ -40,7 +39,32 @@ protected DoubleTensor op(DoubleTensor l, DoubleTensor r) {
     }
 
     @Override
-    protected DualNumber dualOp(DualNumber l, DualNumber r) {
-        return l.div(r);
+    protected PartialDerivatives forwardModeAutoDifferentiation(PartialDerivatives l, PartialDerivatives r) {
+
+        // dc = (B * da - A * db) / B^2;
+        DoubleTensor newValue = this.getValue();
+        PartialDerivatives thisInfMultiplied;
+        PartialDerivatives thatInfMultiplied;
+        PartialDerivatives newInf;
+
+        if (l.isEmpty()) {
+            thisInfMultiplied = PartialDerivatives.OF_CONSTANT;
+        } else {
+            thisInfMultiplied = l.multiplyAlongOfDimensions(right.getValue(), left.getValue().getShape());
+        }
+
+        if (r.isEmpty()) {
+            thatInfMultiplied = PartialDerivatives.OF_CONSTANT;
+        } else {
+            thatInfMultiplied = r.multiplyAlongOfDimensions(left.getValue(), right.getValue().getShape());
+        }
+
+        if (thisInfMultiplied.isEmpty() && thatInfMultiplied.isEmpty()) {
+            newInf = PartialDerivatives.OF_CONSTANT;
+        } else {
+            newInf = thisInfMultiplied.subtract(thatInfMultiplied).divideBy(right.getValue().pow(2));
+        }
```

Comment:
I think this would be a good place to use ` ? : ` syntax

Likewise in Multiplication, Power, MatrixMultiplication vertices